source/server/admin/prometheus_stats.cc

```diff
@@ -274,6 +276,35 @@ uint64_t outputPrimitiveStatType(Buffer::Instance& response, const StatsParams&
   return result;
 }
 
+/*
+ * Returns the prometheus output for a summary. The output is a multi-line string (with embedded
+ * newlines) that contains all the individual quantile values and sum/count for a single histogram
+ * (metric_name plus all tags).
+ */
+std::string generateSummaryOutput(const Stats::ParentHistogram& histogram,
+                                  const std::string& prefixed_tag_extracted_name) {
+  const std::string tags = PrometheusStatsFormatter::formattedTags(histogram.tags());
+  const std::string hist_tags = histogram.tags().empty() ? EMPTY_STRING : (tags + ",");
+
+  const Stats::HistogramStatistics& stats = histogram.cumulativeStatistics();
+  Stats::ConstSupportedBuckets& supported_quantiles = stats.supportedQuantiles();
+  const std::vector<double>& computed_quantiles = stats.computedQuantiles();
+  std::string output;
+  for (size_t i = 0; i < supported_quantiles.size(); ++i) {
+    double quantile = supported_quantiles[i];
+    double value = computed_quantiles[i];
+    output.append(fmt::format("{0}{{{1}quantile=\"{2}\"}} {3}\n", prefixed_tag_extracted_name,
+                              hist_tags, quantile, std::isnan(value) ? 0 : value));
```

Comment:
The spec says that `NaN` is an allowed value (https://prometheus.io/docs/instrumenting/exposition_formats/). Should we leave it as-is here instead of translating it to 0?